be/src/column/map_column.cpp

```diff
@@ -621,4 +621,35 @@ Status MapColumn::unfold_const_children(const starrocks::TypeDescriptor& type) {
     return Status::OK();
 }
 
+// keep the last identical key
+void MapColumn::remove_duplicated_keys() {
+    Filter filter(_keys->size(), 1);
+
+    bool has_duplicated_keys = false;
+    UInt32Column::Ptr new_offsets = UInt32Column::create();
+    auto& offsets_vec = new_offsets->get_data();
+    offsets_vec.push_back(0);
+
+    uint32_t new_offset = 0;
+    for (auto i = 0; i < _offsets->size() - 1; ++i) {
+        for (auto j = _offsets->get_data()[i]; j < _offsets->get_data()[i + 1]; ++j) {
+            for (auto k = j + 1; k < _offsets->get_data()[i + 1]; ++k) {
+                if (_keys->equals(j, *_keys, k)) {
+                    filter[j] = 0;
+                    has_duplicated_keys = true;
+                    break;
+                }
+            }
+            new_offset += filter[j];
+        }
+        offsets_vec.push_back(new_offset);
```

Comment:
can we use hash-based or sort-based algorithm instead of two-layer for-loop?

Reply:
of cause, we can optimize large maps in the future.

Reply:
If there are multiple duplicate keys, it seems that we will choose to keep the last one, is this also the same behavior of the data load phase?